src/file/tokio_compat_file.rs

```diff
@@ -552,13 +633,37 @@ impl AsyncWrite for TokioCompatFile {
 
         let max_write_len = self.max_write_len_impl();
 
-        let (n, bufs, buf) = if let Some(res) = take_io_slices(bufs, max_write_len as usize) {
-            res
-        } else {
+        let Some(n) = take_io_slices(bufs, max_write_len as usize).map(|res| res.0) else {
```

Comment:
I just realized that using `let-else` might bump the msrv:

```suggestion
        let n = if let Some(n) = take_io_slices(bufs, max_write_len as usize).map(|res| res.0) else {
```